internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -408,7 +408,11 @@ func (md *machineDeployment) updateMachineByReplace(ctx context.Context, e *mach
 
 	newMachineRaw, err := md.flapsClient.Launch(ctx, *e.launchInput)
 	if err != nil {
-		return err
+		if strings.Contains(err.Error(), "could not reserve resource for machine") {
+			return errors.New("The region you're trying to deploy is likely at capacity. Consider deploying to a new region with fly deploy --region <region> or trying again in a few hours\n")
```

Comment:
This is a good opportunity to add improved messaging and maybe a link to docs like we talked about in Slack.

Reply:
good point +1

Reply:
For posterity, here's some interfaces we have that lets us tag a custom error type so the downstream error handlers can render them, taking into account a lot more than we can do inside random functions: https://github.com/superfly/flyctl/blob/master/internal/flyerr/flyerr.go#L14-L40

If an error has any meaning to flyctl at all, it needs a custom error type. Even if it's `var CustomError = errors.New("boom")` in a package. That'll let us refer to an error as a type, which makes everything a lot easier.